### PR TITLE
feat(build): add automated ms.date freshness checking

### LINKED ISSUE #164: feat(workflows): add automated ms.date freshness checking for documentation
## Summary

The documentation maintenance policy (`docs/contributing/documentation-maintenance.md`) references automated `ms.date` freshness checking as a planned future enhancement. Implement a GitHub Actions workflow that detects stale `ms.date` frontmatter values in Markdown files and surfaces them during CI or on a schedule.

## Current Behavior

```markdown
> [!NOTE]
> Automated `ms.date` freshness checking (GitHub Actions) is planned for a future milestone.
```

Freshness is tracked manually through the `ms.date` frontmatter field. There is no automated detection of stale documentation.

## Expected Behavior

A GitHub Actions workflow that:

* Parses `ms.date` frontmatter from all Markdown files in the repository
* Flags files where `ms.date` exceeds a configurable staleness threshold (for example, 90 or 180 days)
* Runs on a schedule (for example, weekly) or as part of PR checks
* Outputs a report of stale files with their last `ms.date` value and age

## Acceptance Criteria

* GitHub Actions workflow created in `.github/workflows/`
* Configurable staleness threshold (default 90 days)
* Report output lists file path, `ms.date` value, and days since last update
* Workflow does not block PRs (informational only initially)
* Update the `[!NOTE]` alert in `docs/contributing/documentation-maintenance.md` to reference the workflow once implemented
* Verify `npm run lint:md` passes after any documentation changes

## Context

Identified during PR #162 code review. The current manual approach relies on contributors remembering to update `ms.date`, which is error-prone. Automation improves compliance with the OpenSSF `documentation_current` criterion.

## Files Affected

* `.github/workflows/` (new workflow file)
* `docs/contributing/documentation-maintenance.md` (update NOTE alert)

### PR DESCRIPTION
**Pull request by rezatnoMsirhC**
# feat(build): add automated ms.date freshness checking

## Description

Implemented comprehensive automated documentation freshness checking infrastructure to detect and track stale markdown files based on `ms.date` frontmatter timestamps. The system includes **PowerShell validation scripts** with Pester test coverage, **GitHub Actions workflows** for scheduled scans and PR validation, and **automated issue creation** for stale documentation tracking.

The architecture uses a **dual-orchestrator pattern** separating weekly comprehensive scans (hard-fail mode with GitHub notifications) from PR-scoped validation (soft-fail warnings). Artifact-based data flow connects the validation workflow to the issue automation workflow, creating GitHub issues for each stale file with idempotent duplicate detection via automation markers.

> **Key workflow components:**
> * `msdate-freshness-check.yml` — Reusable workflow with configurable threshold (default: 90 days), soft-fail mode, and changed-files-only filtering
> * `weekly-validation.yml` — Scheduled Monday 9 AM UTC scans with hard-fail mode triggering notifications
> * `pr-validation.yml` — Immediate developer feedback during code review without blocking merges
> * `create-stale-docs-issues.yml` — Issue automation creating/updating GitHub issues using automation markers

The **PowerShell validation script** (`Invoke-MsDateFreshnessCheck.ps1`) parses YAML frontmatter, calculates age in days, generates dual outputs (JSON artifact + markdown summary), and supports changed-files-only mode via Git merge-base comparison. Comprehensive Pester test suite covers file discovery, YAML parsing, report generation, and full workflow integration with CI annotation verification.

**Workflow test results:** Successfully validated the issue creation workflow in [PR #448](https://github.com/Azure-Samples/azure-nvidia-robotics-reference-architecture/actions/runs/22880060875/job/66380653143?pr=448), creating 10 GitHub issues (#513-#522) for stale documentation files including test fixtures (419 days old) and workflow READMEs (96 days old).

Closes #164

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change fixing an issue)
- [x] ✨ New feature (non-breaking change adding functionality)
- [ ] 💥 Breaking change (fix or feature causing existing functionality to change)
- [x] 📚 Documentation update
- [x] 🏗️ Infrastructure change (Terraform/IaC)
- [ ] ♻️ Refactoring (no functional changes)

## Component(s) Affected

- [ ] `deploy/000-prerequisites` - Azure subscription setup
- [ ] `deploy/001-iac` - Terraform infrastructure
- [ ] `deploy/002-setup` - OSMO control plane / Helm
- [ ] `deploy/004-workflow` - Training workflows
- [ ] `src/training` - Python training scripts
- [x] `docs/` - Documentation

## Testing Performed

- [ ] Terraform `plan` reviewed (no unexpected changes)
- [ ] Terraform `apply` tested in dev environment
- [ ] Training scripts tested locally with Isaac Sim
- [ ] OSMO workflow submitted successfully
- [ ] Smoke tests passed (`smoke_test_azure.py`)

**PowerShell Script Validation:**
* **Pester test suite**: All tests passing with comprehensive coverage
  * File discovery and exclusion patterns
  * YAML frontmatter parsing with date format validation
  * Report generation (JSON + markdown output formats)
  * Integration tests covering full processing pipeline
  * CI annotation verification using mock patterns
  * Environment isolation with conditional execution based on PowerShell-Yaml availability

**GitHub Actions Workflow Testing:**
* **Weekly validation workflow**: Created with Monday 9 AM UTC cron schedule and hard-fail mode
* **PR integration**: Added to pr-validation.yml with soft-fail and changed-files-only modes
* **Issue automation workflow**: Successfully tested in [workflow run](https://github.com/Azure-Samples/azure-nvidia-robotics-reference-architecture/actions/runs/22880060875/job/66380653143?pr=448)
  * Created 10 issues for stale documentation (#513-#522)
  * Automation markers correctly implemented for duplicate detection
  * Issue template includes task checklist for documentation updates

**Validation Results:**
* Files checked: 63 markdown files
* Stale files detected: 10 (expected for initial rollout)
  * 8 test fixture files (age: 419 days)
  * 2 workflow READMEs (age: 96 days)
* JSON report generated: `logs/msdate-freshness-results.json` (9.1K)
* Markdown summary generated: `logs/msdate-summary.md` (958 bytes)
* CI warnings emitted correctly via Write-CIAnnotation
* Soft-fail mode prevented exit code 1 as expected

**Linting Validation:**
* **Markdown linting**: 82 files checked, 0 errors
* **Spell checking**: 327 files checked, 0 issues (added `msdate` to custom dictionary)
* **YAML workflow syntax**: Workflows follow repository conventions with action SHA pinning

## Documentation Impact

- [ ] No documentation changes needed
- [x] Documentation updated in this PR
- [ ] Documentation issue filed

**Documentation Changes:**
* Updated `docs/contributing/documentation-maintenance.md`:
  * Replaced "planned for future milestone" note with current implementation details
  * Documented three execution contexts: weekly scans (hard-fail), PR checks (soft-fail), manual dispatch
  * Added "Fixing Stale Documentation" section with remediation workflow
  * Explained ms.date update policy: update on every substantive content edit
  * Refreshed ms.date frontmatter to 2026-03-06
* Updated `.cspell/general-technical.txt`: Added `msdate` to custom dictionary
* Updated `.markdownlint-cli2.jsonc`: Excluded `logs/**` directory from linting to prevent errors on generated workflow outputs

## Bug Fix Checklist

*Complete this section for bug fix PRs. Skip for other contribution types.*

N/A — This is a new feature, not a bug fix.

## Checklist

- [x] My code follows the [project conventions](copilot-instructions.md)
- [x] Commit messages follow [conventional commit format](instructions/commit-message.instructions.md)
- [x] I have performed a self-review
- [x] Documentation impact assessed above
- [x] No new linting warnings introduced

### Execution Contexts

1. **Weekly scans** (weekly-validation.yml): Scheduled for Monday 9 AM UTC, checks all markdown files, uses hard-fail mode (soft-fail: false) to trigger GitHub email notifications when stale files exceed 90-day threshold
2. **PR checks** (pr-validation.yml): Runs automatically on pull requests, checks only modified files (changed-files-only: true), uses soft-fail mode (soft-fail: true) to warn without blocking merges
3. **Manual dispatch**: Available via GitHub Actions UI workflow_dispatch on both weekly-validation.yml and pr-validation.yml

### PowerShell Parameter Binding Debugging Journey

The commit history (30+ commits) documents extensive iterative debugging of PowerShell 7.4.13 array parameter binding issues in GitHub Actions:
* **Root cause**: PowerShell 7.4.13 bug on ubuntu-latest runners with `[string[]]` array defaults
* **Approaches tested**: hashtable splatting, array splatting, string conversion, Invoke-Expression, direct parameter binding
* **Final solution**: Moved parameter defaults outside param block to runtime assignment with `[AllowNull()]` and `[AllowEmptyCollection()]` attributes
* **Reverted commits**: Three commits during iteration (pull_request trigger testing, env var type handling, undefined input handling)

This debugging pattern demonstrates environment-specific edge cases invisible in local development, highlighting the value of CI-based validation.

### Issue Creation Workflow Test Results

**Test workflow run**: https://github.com/Azure-Samples/azure-nvidia-robotics-reference-architecture/actions/runs/22880060875/job/66380653143?pr=448

**Created issues for stale documentation** (10 total):
* #513: `scripts/tests/Fixtures/Frontmatter/empty-description.md` (age: 419 days)
* #514: `scripts/tests/Fixtures/Frontmatter/footer-with-formatting.md` (age: 419 days)
* #515: `scripts/tests/Fixtures/Frontmatter/footer-with-html-comment.md` (age: 419 days)
* #516: `scripts/tests/Fixtures/Frontmatter/valid-docs-with-footer.md` (age: 419 days)
* #517: `scripts/tests/Fixtures/Frontmatter/valid-docs.md` (age: 419 days)
* #518: `scripts/tests/Fixtures/Frontmatter/valid-root-community-with-footer.md` (age: 419 days)
* #519: `scripts/tests/Fixtures/Frontmatter/valid-root-community.md` (age: 419 days)
* #520: `scripts/tests/Fixtures/Frontmatter/warning-author-empty.md` (age: 419 days)
* #521: `workflows/azureml/README.md` (age: 96 days)
* #522: `workflows/osmo/README.md` (age: 96 days)

### Artifact Configuration

* JSON reports retain for 30 days via upload-artifact action
* Markdown summaries append to GITHUB_STEP_SUMMARY for inline visibility in workflow UI
* All generated files save to `logs/` directory (git-ignored, excluded from markdown linting)

### First Scheduled Run

Next Monday at 9:00 AM UTC

### CODE REVIEW

**Comment by rezatnoMsirhC** [comment]
> @rezatnoMsirhC - this looks TIGHT! Nice work!!!! (probably hate me for this ... but would be good to back-port this to hve core, too :-) )

Appreciate it! No problem once this is working I can backport it to hve-core.

File: scripts/linting/Invoke-MsDateFreshnessCheck.ps1 [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+
+        foreach ($file in $sortedStaleFiles) {
+            $markdown += "| $($file.File) | $($file.MsDate) | $($file.AgeDays) |`n"
+        }
+    }
+    else {
+        $markdown += @"
+
+### ✅ All Files Fresh
+ ✅ All Files Fresh
```
Comment (WilliamBerryiii) [inline scripts/linting/Invoke-MsDateFreshnessCheck.ps1]: Nice work on the reporting output! Small catch — this else branch has a duplicate heading: line 212 is `### ✅ All Files Fresh` and line 213 repeats it with a leading space. Looks like the second line may be left over from an earlier edit.

```suggestion
### ✅ All Files Fresh
```
